credd.go

```diff
@@ -0,0 +1,287 @@
+package htcondor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/PelicanPlatform/classad/classad"
+)
+
+// CredType enumerates the credential types supported by the credd.
+// Mirrors the Python bindings CredType enum (Password, Kerberos, OAuth).
+type CredType string
+
+const (
+	// CredTypePassword stores legacy password credentials.
+	CredTypePassword CredType = "Password"
+	// CredTypeKerberos stores Kerberos credentials.
+	CredTypeKerberos CredType = "Kerberos"
+	// CredTypeOAuth stores OAuth2 credentials.
+	CredTypeOAuth CredType = "OAuth"
+)
+
+// ErrCredentialNotFound is returned when the requested credential does not exist.
+var ErrCredentialNotFound = errors.New("credential not found")
+
+// CredentialStatus describes whether a credential exists and when it was last updated.
+type CredentialStatus struct {
+	Exists    bool       `json:"exists"`
+	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+}
+
+// ServiceStatus describes the state of a service credential.
+type ServiceStatus struct {
+	Service   string     `json:"service"`
+	Handle    string     `json:"handle,omitempty"`
+	Exists    bool       `json:"exists"`
+	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+}
+
+// CreddClient defines the operations exposed by the credd (credential daemon).
+// The methods mirror the Python bindings htcondor2.Credd API.
+type CreddClient interface {
+	AddPassword(ctx context.Context, password string, user string) error
+	AddUserCred(ctx context.Context, credType CredType, credential []byte, user string) error
```

Comment:
There seems to be potential overlap between `AddUserCred` and `AddUserServiceCred` (and similar delete / query methods).  It's not clear from the signature which methods to use or if all methods apply to all credential types.

Can you tighten the interface so there are fewer (potentially confusing) options?

While Add/Delete/Get are obvious (though it's unclear why `Get` is specific to OAuth2; would prefer a generic method), it's also less obvious when to use `Query` versus `Check`.

Seems to be missing a "List" method.